File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/brev/FeilmeldingDto.java
package no.nav.melosys.tjenester.gui.dto.brev;

import java.util.Collection;
import java.util.Collections;
import java.util.List;

import no.nav.melosys.domain.kodeverk.begrunnelser.Kontroll_begrunnelser;

public record FeilmeldingDto(String tittel, Collection<FeilmeldingUnderpunkt> underpunkter) {

    public FeilmeldingDto(Kontroll_begrunnelser begrunnelse, String underpunkt) {
        this(begrunnelse.getBeskrivelse(), List.of(new FeilmeldingUnderpunkt(underpunkt)));
    }

    public FeilmeldingDto(Kontroll_begrunnelser begrunnelse) {
        this(begrunnelse.getBeskrivelse(), Collections.emptyList());

    }

    public FeilmeldingDto(String tittel) {
        this(tittel, Collections.emptyList());
    }

}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/brev/FeilmeldingDto.kt
package no.nav.melosys.tjenester.gui.dto.brev

import no.nav.melosys.domain.kodeverk.begrunnelser.Kontroll_begrunnelser
import java.util.List

@JvmRecord
data class FeilmeldingDto(@JvmField val tittel: String, val underpunkter: Collection<FeilmeldingUnderpunkt>) {
    constructor(begrunnelse: Kontroll_begrunnelser, underpunkt: String?) : this(
        begrunnelse.beskrivelse, List.of<FeilmeldingUnderpunkt>(
            FeilmeldingUnderpunkt(
                underpunkt!!
            )
        )
    )

    constructor(begrunnelse: Kontroll_begrunnelser) : this(begrunnelse.beskrivelse, emptyList<FeilmeldingUnderpunkt>())

    constructor(tittel: String?) : this(tittel!!, emptyList<FeilmeldingUnderpunkt>())
}

